btk/measure.py

```diff
@@ -1,119 +1,73 @@
-from abc import ABC
-
+"""Contains examples of functions that can be used to apply a measurement algorithm to the blends
+ simulated by BTK. Every measurement function should take as an input a batch returned from a
+ DrawBlendsGenerator object (see its `__next__` method) and an index corresponding to which image
+ in the batch to measure.
+
+It should return a dictionary containing a subset of the following keys/values:
+    - deblend_image (np.ndarray): Array of deblended isolated images with shape:
+                                  [batch_size, max_sources, n_bands, nx, ny]
+    - peaks (np.ndarray): Array of predicted centroids in pixels (batch_size, max_sources, 2).
+
+Omitted entries are automatically assigned a `None` value.
+"""
 import numpy as np
 import sep
 from skimage.feature import peak_local_max
 
 from btk.multiprocess import multiprocess
 
 
-class MeasurementParams(ABC):
-    """Class with functions to perform detection/deblending/measurement."""
-
-    def make_measurement(self, data, index):
-        """Function describing how the measurement algorithm is run.
+def basic_measure(batch, idx):
+    """Return centers detected when object detection is performed on the
+    input image with skimage.feature.peak_local_max.
 
-        Args:
-            data (dict): Output generated by btk.draw_blends containing blended
-                         images, isolated images, observing conditions and
-                         blend catalog, for a given batch.
-            index (int): Index number of blend scene in the batch to preform
-                         measurement on.
+    NOTE: Assumes at least 3 bands are present in the images produced.
 
-        Returns:
-            output of measurement algorithm (fluxes, shapes, size, etc.) as
-            an astropy catalog.
-        """
-        return None
+    Args:
+        batch (dict): Output of DrawBlendsGenerator object's `__next__` method.
+        idx (int): Index number of blend scene in the batch to preform
+            measurement on.
 
-    def get_deblended_images(self, data, index):
-        """Function describing how the deblending algorithm is run.
+    Returns:
+            dict containing subset of keys from: ['deblend_image', 'peaks']
```

Comment:
I would rename the key in "deblended_images" for consistance. We'll also need to specify which format we want for all of the entries somewhere. Finally, we may want additionnal entries ? I'm thinking about segmentation in particular. Though we can probably handle in the metrics part the case where the key does not exist.

Reply:
Yeah do you have suggestions for specifying the format? I added some stuff in the top of the file but not sure if that is the best place. I could also add assertions checking the output of measurement in the `run_batch` method.

Reply:
I think it should show up in the doc if it is at the top of the file, so it should be alright. Adding checks may help users troubleshooting their problems, this is a good idea.